### PR TITLE
Cleanup of setup.py

### PR DESCRIPTION
**Pull request by anders-kiaer**
- [X] Make dependency definition consistent (i.e. include all in `setup.py`)
- [X] Remove unused dependencies.
- [X] Make `sphinx` build command (or equivalently, `sphinx` installation) optional.
- [X] Add `.nojekyll` file to built documentation.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -2,59 +2,57 @@
...
-
-REQUIREMENTS = parse_requirements("requirements.txt")
-
-SETUP_REQUIREMENTS = ["pytest-runner", "setuptools>=28", "setuptools_scm"]
+SETUP_REQUIREMENTS = ["setuptools>=28", "setuptools_scm"]
 
-TEST_REQUIREMENTS = parse_requirements("requirements_dev.txt")
+TEST_REQUIREMENTS = [
+    "flake8>=2.6.0",
+    "pyarrow",
```
Comment (berland) [inline setup.py]: I think pyarrow must be deleted here, to have travis test both with and without pyarrow.

File: .travis.yml [hunk 2]
```diff
@@ -21,15 +21,16 @@ before_install:
...
   - pip install .
+  - pip install .[tests]
 
 script:
   - python -c "import fmu.ensemble"
-  - python setup.py test
-  - pip install pyarrow
+  - pytest tests/
+  - pip install .[parquet] 
   - pytest -s tests/test_virtualensemble.py
```
Comment (berland) [inline .travis.yml]: `-s` here but not above? Maybe drop it both places.

File: setup.py (hunk 1 above)
Comment (anders-kiaer) [reply setup.py]: Will do (also note that the `dev` requirement list is already a bit shorter).

**Comment by berland** [comment]
https://github.com/equinor/pyscal/blob/61928d0fddb2ac69dd3e59b97d885d4373422d94/.travis.yml#L16  This was needed for pyscal to get the API back again

**Pull request merged**